src/main/java/org/opensearch/knn/index/vectorvalues/KNNMergeVectorValues.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.vectorvalues;
+
+import org.apache.lucene.codecs.KnnVectorsReader;
+import org.apache.lucene.index.ByteVectorValues;
+import org.apache.lucene.index.DocIDMerger;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.FloatVectorValues;
+import org.apache.lucene.index.MergeState;
+import org.apache.lucene.index.VectorEncoding;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.FixedBitSet;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class to get merged VectorValues from MergeState
+ */
+public final class KNNMergeVectorValues {
+
+    /**
+     * Gets list of {@link KNNVectorValuesSub} for {@link FloatVectorValues} from a merge state and returns the iterator which
+     * iterates over live docs from all segments while mapping docIds.
+     *
+     * @param fieldInfo
+     * @param mergeState
+     * @return List of KNNVectorSub
+     * @throws IOException
+     */
+    public static KNNVectorValuesIterator.MergeFloat32VectorValuesIterator mergeFloatVectorValues(
```

Comment:
```suggestion
    public static KNNVectorValuesIterator.MergeFloat32VectorValuesIterator getMergedFloatVectorValuesIterator(
```

Reply:
Changing this to return MergedFloatVectorValues which extends FloatVectorValues so iterator will not make sense